compiler/aarch64/codegen/ControlFlowEvaluator.cpp

```diff
@@ -814,6 +814,24 @@ OMR::ARM64::TreeEvaluator::iselectEvaluator(TR::Node *node, TR::CodeGenerator *c
    TR::Register *falseReg = cg->evaluate(falseNode);
    TR::Register *resultReg = trueReg;
 
+   // Internal pointers cannot be handled since we cannot set the pinning array
+   // on the result register without knowing which side of the select will be
+   // taken.
+   TR_ASSERT_FATAL_WITH_NODE(
+      node,
+      !trueReg->containsInternalPointer() && !falseReg->containsInternalPointer(),
+      "Select nodes cannot have children that are internal pointers"
+   );
+   if (falseReg->containsCollectedReference())
```

Comment:
I don't think this code under the `if` is actually useful on AArch64.  At this point in the function, `resultReg` should be the same as `trueReg`.  `trueReg` and `falseReg` are both the result of evaluation the true and false children, respectively.  So if `falseReg` is a collected reference then `trueReg` better be (and vice versa) otherwise the trees aren't properly formed.

Reply:
The issue is that for NULL it can be collected or not collected so you have to do this copy in case you have a not collected null merging with a collected reference.

Reply:
OK, for some reason I thought that an `aconst 0` evaluated into a register had the collected reference flag set.  It doesn't look like it though.  This change is fine then.